Packs/Carbon_Black_Enterprise_Response/Integrations/CarbonBlackResponseV2/CarbonBlackResponseV2.py

```diff
@@ -0,0 +1,820 @@
+                                         'cb-edr-watchlist-create': watchlist_create_command,
+                                         'cb-edr-watchlist-update': watchlist_update_command,
+                                         'cb-edr-watchlist-delete': watchlist_delete_command,
+                                         'cb-edr-sensors-list': sensors_list_command,
+                                         'cb-edr-quarantine-device': quarantine_device_command,
+                                         'cb-edr-unquarantine-device': unquarantine_device_command,
+                                         'cb-edr-sensor-installer-download': sensor_installer_download_command,
+                                         'fetch-incidents': fetch_incidents,
+                                         'endpoint': endpoint_command
+                                         }
+
+        if command == 'test-module':
+            result = test_module(client)
+            return_results(result)
+
+        elif command == 'fetch-incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            alert_status = demisto.params().get('alert_status', None)
+            alert_feed_name = demisto.params().get('alert_feed_name', None)
+            alert_hostname = demisto.params().get('alert_hostname', None)
+            alert_query = demisto.params().get('alert_query', None)
+
+            # Convert the argument to an int using helper function or set to MAX_INCIDENTS_TO_FETCH
+            max_results = arg_to_number(
+                arg=demisto.params().get('max_fetch'),
+                arg_name='max_fetch',
+                required=False
+            )
+            if not max_results or max_results > MAX_INCIDENTS_TO_FETCH:
+                max_results = MAX_INCIDENTS_TO_FETCH
+            # How much time before the first fetch to retrieve incidents
+            first_fetch_time = arg_to_datetime(
+                arg=demisto.params().get('first_fetch', '3 days'),
+                arg_name='First fetch time',
+                required=True
+            )
+            first_fetch_timestamp = int(first_fetch_time.timestamp()) if first_fetch_time else None
+            # Using assert as a type guard (since first_fetch_time is always an int when required=True)
+            assert isinstance(first_fetch_timestamp, int)
+
+            next_run, incidents = fetch_incidents(
+                client=client,
+                max_results=max_results,
+                last_run=demisto.getLastRun(),  # getLastRun() gets the last run dict
+                first_fetch_time=first_fetch_timestamp,
+                status=alert_status,
+                feedname=alert_feed_name,
+                hostname=alert_hostname,
+                query=alert_query)
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
```

Comment:
please extract this logic to a func that handles the param validation. either in fetch_incidents or in a sub func within it.